File: dsa/topics/18_strings/python/longest_repeating_characters.py
def longest_consecutive_characters(sequence):
    max_count = 0
    max_char = ''

    curr = ''
    prev = ''
    count = 0

    for i in range(len(sequence)):
        curr = sequence[i]

        if prev == curr:
            count += 1
            if count > max_count:
                max_char = curr
                max_count = count
        else:
            count = 1

        prev = curr

    return f"{max_char}:{max_count}"


# ------------------ Test (Same as Java main) ------------------

sequence = "AABCDAADBBBEA"
print(longest_consecutive_characters(sequence))  # Expected: B:3

File: dsa/topics/18_strings/java/LongestRepeatingCharacters.java
public class LongestRepeatingCharacters {

	public static void main(String[] args) {
		LongestRepeatingCharacters obj = new LongestRepeatingCharacters();

		String sequence = "AABCDAADBBBEA";
		System.out.println(obj.longestConsecutiveCharacters(sequence));
	}

	public String longestConsecutiveCharacters(String sequence){

		int maxCount = 0;
		char maxChar = 0;

		char curr = 0, prev = 0;
		int count = 0;
		for(int i=0; i<sequence.length(); i++){
			curr = sequence.charAt(i);

			if(prev == curr) {
				count++;
				if( count > maxCount){
					maxChar = curr;
					maxCount = count;
				}
			}else {
				count = 1;
			}

			prev = curr;
		}

		return maxChar+":"+maxCount;
	}

}
